test/helpers/kubectl.go

```diff
@@ -626,41 +626,79 @@ func (kub *Kubectl) WaitKubeDNS() error {
 	return kub.WaitforPods(KubeSystemNamespace, fmt.Sprintf("-l %s", kubeDNSLabel), DNSHelperTimeout)
 }
 
-// WaitForKubeDNSEntry waits until the given DNS entry is ready in kube-dns
-// pod. If the container is not ready after timeout it returns an error. The
+// WaitForKubeDNSEntry waits until the given DNS entry exists in the kube-dns
+// service. If the container is not ready after timeout it returns an error. The
 // name's format query should be `${name}.${namespace}`. If `svc.cluster.local`
-// is not present it appends to the given name and it checks  the full FQDN
-func (kub *Kubectl) WaitForKubeDNSEntry(name string) error {
+// is not present, it appends to the given name and it checks the service's FQDN.
+func (kub *Kubectl) WaitForKubeDNSEntry(serviceName, serviceNamespace string) error {
 	svcSuffix := "svc.cluster.local"
-	logger := kub.logger.WithField("dnsName", name)
+	logger := kub.logger.WithFields(logrus.Fields{"serviceName": serviceName, "serviceNamespace": serviceNamespace})
 
-	if !strings.HasSuffix(name, svcSuffix) {
-		name = fmt.Sprintf("%s.%s", name, svcSuffix)
+	serviceNameWithNamespace := fmt.Sprintf("%s.%s", serviceName, serviceNamespace)
+	if !strings.HasSuffix(serviceNameWithNamespace, svcSuffix) {
+		serviceNameWithNamespace = fmt.Sprintf("%s.%s", serviceNameWithNamespace, svcSuffix)
 	}
 	// https://bugs.launchpad.net/ubuntu/+source/bind9/+bug/854705
 	digCMD := "dig +short %s @%s | grep -v -e '^;'"
 
+	// Also provide full dig output for debugging purposes.
+	digCmdLong := "dig %s @%s"
+
 	// If it fails we want to know if it's because of connection cannot be
 	// established or DNS does not exist.
 	digCMDFallback := "dig +tcp %s @%s"
 
-	host, _, err := kub.GetServiceHostPort(KubeSystemNamespace, "kube-dns")
+	dnsClusterIP, _, err := kub.GetServiceHostPort(KubeSystemNamespace, "kube-dns")
 	if err != nil {
 		logger.WithError(err).Error("cannot get kube-dns service IP")
 		return err
 	}
 
 	body := func() bool {
-		res := kub.Exec(fmt.Sprintf(digCMD, name, host))
+		serviceIP, _, err := kub.GetServiceHostPort(serviceNamespace, serviceName)
+		if err != nil {
+			log.WithError(err).Errorf("cannot get service IP for service %s", serviceNameWithNamespace)
+			return false
+		}
+
+		// ClusterIPNone denotes that this service is headless; there is no
+		// service IP for this service, and thus the IP returned by `dig` is
+		// an IP of the pod itself, not ClusterIPNone.
+		if serviceIP != v1.ClusterIPNone {
```

Comment:
please, follow always the same format: 
```golang
if serviceIP == v1.ClusterIPNone {
	res := kub.Exec(fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
	if !res.WasSuccessful() {
		_ = kub.Exec(fmt.Sprintf(digCMDFallback, serviceNameWithNamespace, dnsClusterIP))
	}
	res.WasSuccessful() 
}
```

Reply:
I unconditionally want to run 

```
res := kub.Exec(fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
if !res.WasSuccessful() {
    _ = kub.Exec(fmt.Sprintf(digCMDFallback, serviceNameWithNamespace, dnsClusterIP))
}
res.WasSuccessful() 
```

regardless of it's a `ClusterIP` or not so I don't understand the rationale behind what you are saying here.